driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/internal/BatchCursorFluxTest.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.reactivestreams.client.internal;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.event.CommandEvent;
+import com.mongodb.internal.connection.TestCommandListener;
+import com.mongodb.reactivestreams.client.FindPublisher;
+import com.mongodb.reactivestreams.client.MongoClient;
+import com.mongodb.reactivestreams.client.MongoClients;
+import com.mongodb.reactivestreams.client.MongoCollection;
+import com.mongodb.reactivestreams.client.TestSubscriber;
+import org.bson.Document;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.mongodb.ClusterFixture.TIMEOUT_DURATION;
+import static com.mongodb.ClusterFixture.getDefaultDatabaseName;
+import static com.mongodb.reactivestreams.client.Fixture.drop;
+import static com.mongodb.reactivestreams.client.Fixture.getMongoClientBuilderFromConnectionString;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertArrayEquals;
+
+public class BatchCursorFluxTest {
+
+    private MongoClient client;
+    private TestCommandListener commandListener;
+    private MongoCollection<Document> collection;
+
+    @Before
+    public void setUp() {
+        commandListener = new TestCommandListener(singletonList("commandStartedEvent"), asList("insert", "killCursors"));
+        MongoClientSettings mongoClientSettings = getMongoClientBuilderFromConnectionString().addCommandListener(commandListener).build();
+        client = MongoClients.create(mongoClientSettings);
+        collection = client.getDatabase(getDefaultDatabaseName()).getCollection(getClass().getName());
+        drop(collection.getNamespace());
+    }
+
+    @After
+    public void tearDown() {
+        if (collection != null) {
+            drop(collection.getNamespace());
+        }
+        if (client != null) {
+            client.close();
+        }
```

Comment:
Do we want to do `try{drop} finally {close}`, which has more chances to close the client if things go wrong?

Reply:
Done